years/2020/04/solutions.ts

```diff
@@ -0,0 +1,75 @@
+export const part1 = (input: string[]) => {
+  return input.filter(val => {
+    const passwordObj = passwordObject(val)
+
+    if (!passwordHasValidKeys(passwordObj)) {
+      return false
+    }
+
+    return true
+  }).length
+}
+
+export const part2 = (input: string[]) => {
+  return input.filter(val => {
+    const passwordObj = passwordObject(val)
+
+    if (!passwordHasValidKeys(passwordObj)) {
+      return false
+    }
+
+    return Object.keys(passwordObj).every(key => {
+      const value = passwordObj[key]
```

Comment:
You could use `Object.entries` if you're interested in both the key and the value

```suggestion
    return Object.entries(passwordObj).every(([ key, value ]) => {
```